src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -495,41 +495,16 @@ private static final class ProxyBuilder {
         private static final ClassLoaderValue<Boolean> reverseProxyCache =
             new ClassLoaderValue<>();
 
-        private static Class<?> defineProxyClass(Module m, List<Class<?>> interfaces) {
-            String proxyPkg = null;     // package to define proxy class in
-            int accessFlags = Modifier.PUBLIC | Modifier.FINAL;
-            boolean nonExported = false;
+        private record ProxyContext(Module module, String pkg, boolean packagePrivate) {}
```

Comment:
This represents the context for a proxy class.   It seems `ProxyClassContext` would be clearer.   I suggest the context to have the accessFlags for a proxy class rather than the boolean whether it's package private.   The constructor should check that it must be 0 or `Modifier.PUBLIC`.   `FINAL` will be added to the access flags when it generates the proxy class.